test/GitcoinGovernor.t.sol

```diff
@@ -427,3 +428,482 @@ contract GitcoinGovernorAlphaPostProposalTest is GitcoinGovernorProposalTestHelp
     governorAlpha.queue(_newProposalId);
   }
 }
+
+contract NewGitcoinGovernorProposalTest is GitcoinGovernorProposalTestHelper {
+  // From GovernorCountingSimple
+  uint8 constant AGAINST = 0;
+  uint8 constant FOR = 1;
+  uint8 constant ABSTAIN = 2;
+
+  function assumeReceiver(address _receiver) public {
+    // We don't want the receiver to be the Timelock, as that would make our
+    // assertions less meaningful -- most of our tests want to confirm that
+    // proposals can cause tokens to be sent *from* the timelock to somewhere
+    // else. We also don't want the zero address as a receiver -- if our
+    // governor is sending tokens to the zero address it could just be due to a
+    // bug.
+    vm.assume(_receiver != TIMELOCK && _receiver != address(0x0));
+  }
+
+  function buildProposalData(string memory _signature, bytes memory _calldata)
+    public
+    pure
+    returns (bytes memory)
+  {
+    return abi.encodePacked(bytes4(keccak256(bytes(_signature))), _calldata);
+  }
+
+  function jumpToActiveProposal(uint256 _proposalId) public {
+    uint256 _snapshot = governorBravo.proposalSnapshot(_proposalId);
+    vm.roll(_snapshot + 1);
+  }
+
+  function jumpToVotingComplete(uint256 _proposalId) public {
+    // Jump one block past the proposal voting deadline
+    uint256 _deadline = governorBravo.proposalDeadline(_proposalId);
+    vm.roll(_deadline + 1);
+  }
+
+  function jumpPastProposalEta(uint256 _proposalId) public {
+    uint256 _eta = governorBravo.proposalEta(_proposalId);
+    vm.roll(block.number + 1);
+    vm.warp(_eta + 1);
+  }
+
+  function delegatesVoteOnBravoGovernor(uint256 _proposalId, uint8 _support) public {
+    require(_support < 3, "Invalid value for support");
```

Comment:
Is a `require` the right thing to use here? Genuinely asking because I've never thought to use one in a test before, and I've generally opted for asserts instead, but off the top of my head I don't see a reason why `require` wouldn't work too.

Reply:
See https://github.com/gitcoinco/2022-Governor-upgrade/pull/12#discussion_r1071456984 — basically the test assertion helpers don't stop execution of a test when an assertion fails, but in this case (if support >= 3) something went wrong and we don't care about the rest of the test, so it'd just clutter output. In other words, we're using `require` because this is input verification, and not a test assertion